Theme upgrade modals: Change tier names

dsas opened the pull request:
Make some of the theme tier names align with the plans required for them and upgrade modals to match.

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #86482

## Proposed Changes

Change the tier names for the personal and premium plans in the modal. I've purposefully changed the heading to "Unlock this theme" so that it fits on one line which looks nicer and also to reduce the repetition of Starter/Explorer everywhere.

I also considered making the Explorer themes modal list "Starter themes" in the plan benefits, but thought maybe that's just too confusing and isn't exactly related to the aim of "changing tier names".

## Testing Instructions

### Modals
1. Use the calypso live link
2. Go through to the design picker using a free plan
3. Reload the page after appending flags=themes/tiers to the url
4. Choose a Starter theme like Epi, press "Unlock theme"
5. Choose an Explorer theme like Annalee, press "Unlock theme"

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

Theme type | Before | After
----|---|------
Starter | <img width="821" alt="Screenshot 2024-01-17 at 09 26 06" src="https://github.com/Automattic/wp-calypso/assets/93301/cf52659c-6d3e-474a-ad7c-c9135534a6e4">  | <img width="803" alt="Screenshot 2024-01-17 at 11 53 46" src="https://github.com/Automattic/wp-calypso/assets/93301/8e4733c5-7792-43c0-b0c6-dbe79e9a6768">
Explorer | <img width="817" alt="Screenshot 2024-01-17 at 09 32 48" src="https://github.com/Automattic/wp-calypso/assets/93301/84657602-dee5-427b-8173-fb0ec2edfd43">  | <img width="803" alt="Screenshot 2024-01-17 at 11 53 09" src="https://github.com/Automattic/wp-calypso/assets/93301/3ed86301-69dc-4166-a6a4-b1ba8f8635c9">

### Plans grid
1. Use the calypso live link
2. Go to /plans/:siteslug

Before | After
-------|------
<img width="860" alt="Screenshot 2024-01-17 at 12 02 15" src="https://github.com/Automattic/wp-calypso/assets/93301/843ef53e-0185-4ff3-a2ac-c0a6f54714c2"> |<img width="860" alt="Screenshot 2024-01-17 at 12 01 42" src="https://github.com/Automattic/wp-calypso/assets/93301/c3a0898d-f88b-445f-a3b0-413643667753">





## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

dsas committed:
Theme upgrade modals: Change tier names

Make some of the theme tier names align with the plans required for them.

dsas committed:
Update the features list for personal themes

dsas committed:
Update premium themes plan feature too

dsas commented on client/lib/plans/features-list.tsx:
```diff
@@ -1958,8 +1958,8 @@ export const FEATURES_LIST: FeatureList = {
 		getSlug: () => FEATURE_PREMIUM_THEMES_V2,
-		getTitle: () => i18n.translate( 'Premium themes' ),
-		getIcon: () => <img src={ Theme2Image } alt={ i18n.translate( 'Premium themes' ) } />,
+		getTitle: () => i18n.translate( 'Explorer themes' ),
```
This change will affect the the plans grid. 

I haven't changed the title/description as it's using premium in a more descriptive sense and is used to supplement the "Explorer themes" label, e.g. in alt text.

rcrdortiz replied:
If this affects the Plans Grid, I think that we should add the `String Freeze` label to this PR. PaYJgx-3sX-p2

dsas replied:
yeah I was anticipating some feedback on the strings so I was holding off, and lo and behold!

rcrdortiz replied:
When we merge these changes the Plans Grid will show the `Starter Themes` feature regardless of if we shipped the new tier or not. I think it's already happening in production though.

dsas replied:
We don't include Starter themes in the plans grid with or without this change. We do include "Premium themes" and that will change to "Explorer themes" with this change.

We can revaluate when translations are ready whether we need to adjust before deploying.

dsas commented on client/lib/plans/features-list.tsx:
```diff
@@ -541,10 +541,10 @@ export const FEATURES_LIST: FeatureList = {
 		getDescription: () =>
 			i18n.translate(
-				'Unlimited access to all of our personal themes, including designs specifically tailored for businesses.'
+				'Unlimited access to all of our starter themes, including designs specifically tailored for businesses.'
```
I don't want to block this PR on it, but the business mention surprised me. I don't think the Starter plan is generally the best fit there. Something for another day.

rcrdortiz replied:
Yup, the business reference is strange. 
Should we implement something so that we can retrieve the `{you name it} themes` as we do with the plan names? That way we would only need to change the `getTitle` implementation instead of searching and replacing all occurences. We don't need to do this in this PR, we could do it as boyscout work. 
WDYT?

Or couldn't we just replace this with a similar approach to the one we use on line 172?
```
'Get access to our %(premiumPlanName)s themes, and a...
```

dsas replied:
The difference is case, it would say "Get access to our Starter themes", which might be the right thing to do, but isn't what we were doing before.

I think we should probably put this PR on hold until paYJgx-4vW-p2 is resolved.

rcrdortiz commented on client/components/theme-upgrade-modal/index.tsx:
```diff
@@ -120,14 +120,12 @@ export const ThemeUpgradeModal = ( {
-				<h1 className="theme-upgrade-modal__heading">
-					{ translate( 'Unlock this personal theme' ) }
-				</h1>
+				<h1 className="theme-upgrade-modal__heading">{ translate( 'Unlock this theme' ) }</h1>
```
Nice that we're going with generic reference instead of explicit, will save us time if we ever change plan names/tiers again.

rcrdortiz commented on client/components/theme-upgrade-modal/index.tsx:
```diff
@@ -120,14 +120,12 @@ export const ThemeUpgradeModal = ( {
 				<p>
 					{ translate(
-						'Get access to Personal themes, and a ton of other features, with a subscription to the %(plan)s plan. It’s {{strong}}%(planPrice)s{{/strong}} a year, risk-free with a 14-day money-back guarantee.',
+						'Get access to %(plan)s themes, and a ton of other features, with a subscription to the %(plan)s plan. It’s {{strong}}%(planPrice)s{{/strong}} a year, risk-free with a 14-day money-back guarantee.',
```
I'm seeing that this relies on `plan: plans.data?.[ PLAN_PERSONAL ]?.productNameShort || ''` at line 135.
Do you know the difference between that and `getPlan( PLAN_PERSONAL )?.getTitle()`.

Should we try to always use the same approach?

dsas replied:
`usePlans` (where `plans` here comes from) uses a rest endpoint. `getPlan` is backed by the `PLANS_LIST` constant.

I've just tested this modal on live preview and it appears to be broken in a similar way to #86411 though, so I guess it needs to use getPlan.

dsas replied:
Even though with the current string nothing has changed, it's still broken on my sandbox and working on prod.

dsas replied:
Opened https://github.com/Automattic/wp-calypso/pull/86579 to deal with this - it's broken on trunk too.

rcrdortiz commented on client/components/theme-upgrade-modal/index.tsx:
```diff
@@ -166,14 +164,12 @@ export const ThemeUpgradeModal = ( {
 				<p>
 					{ translate(
-						'Get access to our Premium themes, and a ton of other features, with a subscription to the %(premiumPlanName)s plan. It’s {{strong}}%(planPrice)s{{/strong}} a year, risk-free with a 14-day money-back guarantee.',
+						'Get access to our %(premiumPlanName)s themes, and a ton of other features, with a subscription to the %(premiumPlanName)s plan. It’s {{strong}}%(planPrice)s{{/strong}} a year, risk-free with a 14-day money-back guarantee.',
```
Same as my other comment about how we fetch the plan name.

dsas committed:
Implement agreed modal wording

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/10826520

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @dsas for including a screenshot in the description! This is really helpful for our translators.

rcrdortiz reviewed (approved):
Besides that this will be available to everyone once merged, everything is working as expected.

dsas committed:
Lock wording changes that affect the plans grid behind feature flag and
translation check

dsas commented on client/lib/plans/features-list.tsx:
```diff
@@ -1958,8 +1958,18 @@ export const FEATURES_LIST: FeatureList = {
+			const localeSlug = i18n.getLocaleSlug();
+			const shouldShowNewString =
+				isEnabled( 'themes/tiers' ) &&
+				( ( localeSlug && config< string >( 'english_locales' ).includes( localeSlug ) ) ||
```
this breaks in live previews (wpcom-block-editor) with "Could not find config value for key 'english_locales'"

I'll just remove it and rely on the feature flag.

dsas committed:
Simplify translation checking

The english locales config isn't available in wpcom-block-editor, and
these strings are used in the theme upgrade modal in that context.

dsas commented:
Dropped the string freeze tag, since the plans grid change is now behind the hasTranslation check and many of the other changes are behind feature flags.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/10826520).
